File: master/font_vue/src/api/homework.js
import request from './request'

export const homeworkApi = {
  // === 教师端 ===
  // 自有作业/试卷列表
  getOwnHomework: (params) => request.get('/teacher/homework/own', { params }),

  // 云作业/试卷列表
  getCloudHomework: (params) => request.get('/teacher/homework/cloud', { params }),

  // 我的组卷列表
  getMyPapers: (params) => request.get('/teacher/homework/papers', { params }),

  // 章节目录
  getChapters: (homeworkId) => request.get(`/teacher/homework/${homeworkId}/chapters`),

  // 布置作业
  assignHomework: (data) => request.post('/teacher/homework/assign', data),

  // === 学生端 ===
  // 最新作业
  getLatestHomework: (params) => request.get('/student/homework/latest', { params }),

  // 历次作业
  getHistoryHomework: (params) => request.get('/student/homework/history', { params }),

  // 错题集
  getErrors: (params) => request.get('/student/homework/errors', { params }),

  // 提交作业
  submitHomework: (id, data) => request.post(`/student/homework/${id}/submit`, data),

  // 作业批改结果
  getCorrection: (id) => request.get(`/student/homework/${id}/correction`),

  // 作业报告
  getReport: (id) => request.get(`/student/homework/${id}/report`)
}

File: master/after_fastapi/app/routers/student/homework.py
import uuid

from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.dependencies import get_db, get_current_user, require_role
from app.models.user import User
from app.schemas.homework import SubmitHomeworkRequest
from app.services.homework_service import StudentHomeworkService

router = APIRouter(prefix="/homework", tags=["学生端-作业管理"])


@router.get("/latest")
async def get_latest(
    page: int = Query(default=1, ge=1),
    pageSize: int = Query(default=15, ge=1, le=100),
    current_user: User = Depends(require_role("student")),
    db: AsyncSession = Depends(get_db),
):
    """最新作业"""
    service = StudentHomeworkService(db)
    return await service.get_latest(current_user.id, page, pageSize)


@router.get("/history")
async def get_history(
    page: int = Query(default=1, ge=1),
    pageSize: int = Query(default=15, ge=1, le=100),
    book: str | None = Query(default=None),
    name: str | None = Query(default=None),
    time: str | None = Query(default=None),
    sortField: str | None = Query(default=None),
    sortOrder: str | None = Query(default=None),
    current_user: User = Depends(require_role("student")),
    db: AsyncSession = Depends(get_db),
):
    """历史作业"""
    service = StudentHomeworkService(db)
    return await service.get_history(
        current_user.id, page, pageSize,
        book=book, name=name, time=time,
        sort_field=sortField, sort_order=sortOrder,
    )


@router.get("/errors")
async def get_errors(
    page: int = Query(default=1, ge=1),
    pageSize: int = Query(default=15, ge=1, le=100),
    name: str | None = Query(default=None),
    book: str | None = Query(default=None),
    startTime: str | None = Query(default=None),
    endTime: str | None = Query(default=None),
    sortField: str | None = Query(default=None),
    sortOrder: str | None = Query(default=None),
    current_user: User = Depends(require_role("student")),
    db: AsyncSession = Depends(get_db),
):
    """错题集"""
    service = StudentHomeworkService(db)
    return await service.get_errors(
        current_user.id, page, pageSize,
        name=name, book=book,
        start_time=startTime, end_time=endTime,
        sort_field=sortField, sort_order=sortOrder,
    )


@router.post("/{homework_id}/submit")
async def submit_homework(
    homework_id: uuid.UUID,
    req: SubmitHomeworkRequest,
    current_user: User = Depends(require_role("student")),
    db: AsyncSession = Depends(get_db),
):
    """提交作业"""
    service = StudentHomeworkService(db)
    return await service.submit(current_user.id, homework_id, req.content)


@router.get("/{homework_id}/correction")
async def get_correction(
    homework_id: uuid.UUID,
    current_user: User = Depends(require_role("student")),
    db: AsyncSession = Depends(get_db),
):
    """批改结果"""
    service = StudentHomeworkService(db)
    return await service.get_correction(current_user.id, homework_id)


@router.get("/{homework_id}/report")
async def get_report(
    homework_id: uuid.UUID,
    current_user: User = Depends(require_role("student")),
    db: AsyncSession = Depends(get_db),
):
    """作业报告"""
    service = StudentHomeworkService(db)
    return await service.get_report(current_user.id, homework_id)
